conans/client/graph/graph_builder.py

```diff
@@ -222,8 +222,16 @@ def _expand_require(self, require, node, graph, check_updates, update,
             self._resolve_cached_alias([require], graph)
             # As we are closing a diamond, there can be conflicts. This will raise if conflicts
             conflict = self._conflicting_references(previous.ref, require.ref, node.ref)
-            if conflict:
-                raise ConanException(conflict)
+            if conflict:  # It is possible to get conflict from alias, try to resolve it
+                conflicting_ref = require.ref
+                self._resolve_recipe(node, graph, require, check_updates,
+                                     update, remotes, profile_host, graph_lock)
+                if conflicting_ref == require.ref:  # nothing has changed
+                    raise ConanException(conflict)
+                # if require.ref has changed, it was an ALIAS, so we can check conflict again
+                conflict = self._conflicting_references(previous.ref, require.ref, node.ref)
+                if conflict:
+                    raise ConanException(conflict)
```

Comment:
I would keep it more straightforward:

```python
            if conflict:  # It is possible to get conflict from alias, try to resolve and check again
                self._resolve_recipe(node, graph, require, check_updates,
                                     update, remotes, profile_host, graph_lock)
                conflict = self._conflicting_references(previous.ref, require.ref, node.ref)
                if conflict:
                    raise ConanException(conflict)
```

Reply:
Actually I had exactly that initially, but somehow I felt that checking if they were conflicting if I knew nothing changed was bad, but I agree, lets recover it.